File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.enums.notification_type import NotificationType


class NotificationRead(BaseModel):
    """Resposta de `GET /api/v1/notifications`
    (contracts/dashboard-and-notifications.md)."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    type: NotificationType
    title: str
    message: str
    task_id: uuid.UUID | None
    is_read: bool
    created_at: datetime

File: frontend/src/types/notification.ts
export type NotificationType = "DUE_SOON" | "NEW_COMMENT" | "TASK_CHANGED";

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  task_id: string | null;
  is_read: boolean;
  created_at: string;
}
